components/experimental/src/dimension/provider/currency_patterns.rs

```diff
@@ -17,11 +17,58 @@ use icu_provider::prelude::*;
 #[icu_provider::data_struct(marker(CurrencyPatternsDataV1Marker, "currency/patterns@1"))]
 #[derive(Debug, Clone, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-#[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
-#[cfg_attr(feature = "datagen", databake(path = icu_experimental::dimension::provider::currency_patterns))]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
 #[yoke(prove_covariance_manually)]
 pub struct CurrencyPatternsDataV1<'data> {
     /// Contains the unit patterns for a currency based on plural rules.
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub patterns: PluralPatterns<'data, DoublePlaceholder>,
 }
+
+impl<'data> CurrencyPatternsDataV1<'data> {
+    /// Construct an instance directly from a byte slice.
+    ///
+    /// # Safety
+    ///
+    /// The bytes must have been returned by [`Self::as_byte_slice`].
+    pub const unsafe fn from_byte_slice_unchecked(bytes: &'data [u8]) -> Self {
+        Self {
+            patterns: PluralPatterns {
+                strings: icu_plurals::provider::PluralElementsPackedCow {
+                    elements: alloc::borrow::Cow::Borrowed(
+                        icu_plurals::provider::PluralElementsPackedULE::from_byte_slice_unchecked(
+                            bytes,
```

Comment:
issue: Going to merge to fix CI, but  generally I'd like to follow the crabal guidelines of every line of unsafe being strongly justified: here this would mean saying that "from_byte_slice_unchecked requires that the bytes come from a valid byte slice, which the caller invariants enforce by requiring callers to use as_byte_slice" and also document as_byte_slice with "safety usable invariant: returns the slyce of the internal elements PluralElementsPackedULE" or something

Reply:
done